spec/components/input/InputDecorator.spec.tsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import userEvent from '@testing-library/user-event';
+import {
+  fireEvent,
+  render,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+} from '@testing-library/react';
+
+import { InputDecorator, Validation } from '../../../src/components';
+import { Validators } from '../../../src/core/validators/validators';
+
+describe('InputDecorator Component', () => {
+  describe('InputDecorator test suite => ', () => {
+    afterEach(() => {
+      jest.restoreAllMocks();
+    });
+
+    it('render a default input', () => {
+      render(
+        <InputDecorator>
+          <input />
+        </InputDecorator>
+      );
+      const input = screen.getByRole('textbox');
+      expect(input).toBeDefined();
+
+      // An ID should be generated
+      expect(input.id).toBeDefined();
+
+      // 'tk-input' should be added
+      expect(input.classList).toContain('tk-input');
+    });
+    it('render an input with an ID and some CSS classes', () => {
+      const id = 'test-id';
+      const cssClass1 = 'test-class-1';
+      const cssClass2 = 'test-class-1';
+      render(
+        <InputDecorator>
+          <input id={id} className={classnames(cssClass1, cssClass2)} />
+        </InputDecorator>
+      );
+      const input = screen.getByRole('textbox');
+      expect(input).toBeDefined();
+
+      // ID should be kept
+      expect(input.id).toBe(id);
+
+      // 'tk-input' should be added to the existing classes
+      expect(input.classList).toContain('tk-input');
+      expect(input.classList).toContain(cssClass1);
+      expect(input.classList).toContain(cssClass2);
+    });
+    it('throw an error when rendering an invalid child tag (button)', () => {
+      const spy = jest.spyOn(console, 'error').mockImplementation(() => {
+        return;
+      });
+      expect(() =>
+        render(
+          <InputDecorator>
+            <button>A button</button>
+          </InputDecorator>
+        )
+      ).toThrowError(
+        'The Input decorator accepts only an "input" tag. Found: button'
```

Comment:
personnally i'd not match the exact string returned by the error, a simple change in the sentence would break the test